pkgs/tools/security/john/default.nix

```diff
@@ -1,5 +1,6 @@
 { lib, stdenv, fetchFromGitHub, openssl, nss, nspr, libkrb5, gmp, zlib, libpcap, re2
-, gcc, python3Packages, perl, perlPackages, makeWrapper, }:
+, gcc, python3Packages, perl, perlPackages, opencl-headers, ocl-icd, substituteAll, makeWrapper
```

Comment:
Make this optional, [like this](https://github.com/NixOS/nixpkgs/blob/690518e96630dc350aef116f89c1b9f29c3bb9bc/pkgs/by-name/wa/waybar/package.nix#L49)

Reply:
OK, done.

Reply:
> Can default to true, if this does not modify default behaviour. Non-opencl usage should work fine.

Yes, it doesn't modify the default behavior. You have to explicitly request the OpenCL versions of algorithms. Reference: https://github.com/openwall/john/blob/6be5461b67951290f707021b93b34ea4168fa9e2/doc/README-OPENCL#L93-L94

> Please also confirm if it works normally as it is, (ie, without the OpenCL drivers exposed in `run/opengl-drivers` by using `hardware.graphics` option or some sort)

Yes, it works as expected. Non-OpenCL functionality works as before. Requesting OpenCL functionality (e.g. with `--list=opencl-devices`) results in an error:

```
Error: No OpenCL-capable platforms were detected by the installed OpenCL driver.
Error: No OpenCL-capable devices were detected by the installed OpenCL driver.
```

(I tested without `OCL_ICD_VENDORS` on a non-NixOS machine.)